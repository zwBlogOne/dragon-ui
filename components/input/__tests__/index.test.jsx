import React from 'react';
import { render, mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../index';

describe('Input', () => {
  it('renders different types of Input correctly', () => {
    const wrapper = render(
      <div>
        <Input type="text" placeholder="请输入" />
        <Input type="textarea" placeholder="请输入" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders different sizes of Input correctly', () => {
    const wrapper = render(
      <div>
        <Input size="xl" placeholder="请输入" />
        <Input size="lg" placeholder="请输入" />
        <Input size="sm" placeholder="请输入" />
        <Input size="xs" placeholder="请输入" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Input with radius correctly', () => {
    const wrapper = render(
      <div>
        <Input radius type="text" placeholder="请输入" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Disabled Input correctly', () => {
    const wrapper = render(
      <div>
        <Input disabled type="text" value="禁用" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with initial value', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <div>
        <Input onChange={onChange} value="hey" type="text" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when changing Input value', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <div>
        <Input onChange={onChange} type="text" />
      </div>
    );

    wrapper.find('input').simulate('change');
    expect(onChange).toHaveBeenCalled();
  });
});
