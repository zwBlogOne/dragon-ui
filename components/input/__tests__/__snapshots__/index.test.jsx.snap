// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders Disabled Input correctly 1`] = `
<div>
  <span>
    <input
      class="ui-input disabled"
      disabled=""
      type="text"
      value="禁用"
    />
  </span>
</div>
`;

exports[`Input renders Input with radius correctly 1`] = `
<div>
  <span>
    <input
      class="ui-input radius"
      placeholder="请输入"
      type="text"
    />
  </span>
</div>
`;

exports[`Input renders correctly with initial value 1`] = `
<div>
  <Input
    onChange={[Function]}
    prefixCls="ui-input"
    size={null}
    type="text"
    value="hey"
  />
</div>
`;

exports[`Input renders different sizes of Input correctly 1`] = `
<div>
  <span>
    <input
      class="ui-input size-xl"
      placeholder="请输入"
      type="text"
    />
  </span>
  <span>
    <input
      class="ui-input size-lg"
      placeholder="请输入"
      type="text"
    />
  </span>
  <span>
    <input
      class="ui-input size-sm"
      placeholder="请输入"
      type="text"
    />
  </span>
  <span>
    <input
      class="ui-input size-xs"
      placeholder="请输入"
      type="text"
    />
  </span>
</div>
`;

exports[`Input renders different types of Input correctly 1`] = `
<div>
  <span>
    <input
      class="ui-input"
      placeholder="请输入"
      type="text"
    />
  </span>
  <span>
    <textarea
      class="ui-input"
      placeholder="请输入"
    />
  </span>
</div>
`;
